### ISSUE TITLE
Invalid rust syntax in generated constants

### ISSUE THREAD

**Issue by Michael-F-Bryan** [servo/rust-bindgen#582]
### Input C/C++ Header

```C
#include <r_lib.h>
```

### Bindgen Invokation

```rust
extern crate bindgen;

use std::env;
use std::path::Path;

fn main() {
    let out_dir = env::var("OUT_DIR").unwrap();
    let _ = bindgen::builder()
        .header("bindings.h")
        .clang_arg("-I/usr/include/libr/")
        .generate()
        .unwrap()
        .write_to_file(Path::new(&out_dir).join("ffi.rs"));
}
```

### Actual Results

```text
$ cargo build
   Compiling radare2-sys v0.1.0 (file:///home/michael/Documents/radare2-sys)
error: unexpected token: `;`
   --> /home/michael/Documents/radare2-sys/target/debug/build/radare2-sys-53f778355b791841/out/ffi.rs:126:31
    |
126 | pub const INFINITY: f32 = inf.;
    |                               ^

error: unexpected token: `;`
   --> /home/michael/Documents/radare2-sys/target/debug/build/radare2-sys-53f778355b791841/out/ffi.rs:127:26
    |
127 | pub const NAN: f32 = NaN.;
    |                          ^

error: aborting due to 2 previous errors

error: Could not compile `radare2-sys`.
```

### Expected Results

Bindgen shouldn't generate invalid syntax when declaring the two 32-bit float constants. That exact same `C` header file compiles fine when using Clang:

```bash
$ clang -I/usr/include/libr bindings.h
```

I am trying to generate bindings for [radare2][radare2]. `radare2` was installed from source and its include files are put in `/usr/include/libr`.

### `RUST_LOG=bindgen` Output

(output is identical to `Actual Results` section)

### Miscellaneous Other Details

rustc: `rustc 1.17.0-nightly (6f10e2f63 2017-03-14)`
cargo: `cargo 0.18.0-nightly (4a3c0a63b 2017-03-12)`
bindgen: `0.22.1`
llvm: `3.9.1`
target triple: `x86_64-unknown-linux-gnu`
OS: `Arch Linux`

[radare2]: https://github.com/radare/radare2

**Comment by emilio** [comment]
Reduced test case:

```c
/* portable non-c99 inf/nan types */
#if !defined(INFINITY)
#define INFINITY (1.0f/0.0f)
#endif

#if !defined(NAN)
#define NAN (0.0f/0.0f)
#endif
```

**Comment by Michael-F-Bryan** [comment]
@emilio, thanks for simplifying the test case. I haven't looked through the `radare2` source code yet, so I wasn't sure whether they came from a basic `#define NAN (0.0f/0.0f)` or if it was something more complex than that.

**Comment by emilio** [comment]
No worries! Fix over #583.
